hooks/tk-multi-publish2/basic/publish_lookfiles.py

```diff
@@ -0,0 +1,426 @@
+                if os.path.exists(publish_path) and publish_fields["version"] == work_fields["version"]:
+                    self.logger.debug("'{}' already published".format(work_path))
+                    # publish version exists
+                    work_paths_to_publish.remove(work_path)
+                    break
+
+        if not work_paths_to_publish:
+            self.logger.debug(
+                "'{}': All versions published".format(item.name)
+            )
+            return {"accepted": False, "visible": True, "enabled": True, "checked": False}
+
+        settings_dict = {
+            "work_paths": work_paths_to_publish,
+            "to_publish": work_paths_to_publish[0],
+        }
+        self._set_item_settings(settings_dict, settings)
+
+        return {
+            "accepted": True,
+            "checked": True
+        }
+    
+    def _get_template(self, template_name):
+        engine = sgtk.platform.current_engine()
+        return engine.get_template_by_name(template_name)
+
+    def validate(self, settings, item):
+        """
+        Validates the given item to check that it is ok to publish. Returns a
+        boolean to indicate validity.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+        :returns: True if item is valid, False otherwise.
+        """
+        item_settings = self._get_item_settings(settings)
+        # Check the templates exist
+        work_template = item.properties["work_template"]
+        if not work_template:
+            raise Exception("'{}': Work template '{}' doesn't exist".format(item.name, work_template.name))
+        publish_template = item.properties["publish_template"]
+        if not publish_template:
+            raise Exception("'{}': Publish template '{}' doesn't exist".format(item.name, publish_template.name))
+        # Check the file exists still. It might have been removed from the filesystem after collection
+        path = item_settings["to_publish"]
+        if not os.path.exists(path):
+            raise Exception("The file '{}' no longer exists on disk. Has it been deleted/published?".format(path))
+        # Check the filepath is valid
+        if not work_template.validate(path):
+            raise Exception("The filepath '{}' does not match the template '{}'".format(path, work_template.name))
+        # Check if file has already been copied to the publish location
+        fields = work_template.validate_and_get_fields(path)
+        publish_path = publish_template.apply_fields(fields)
+        item.properties["publish_path"] = publish_path
+        item.properties["path"] = path
+        if os.path.exists(publish_path):
+            raise Exception("The file '{}' has already been copied to the publish location.".format(path))
```

Comment:
Switch to use explicit `IOError`

```suggestion
            import errno
            raise IOError(errno.EEXIST, "The file has already been copied to the publish location:", path)
```